### ISSUE TITLE
Cheats not working

### ISSUE THREAD

**Issue by Phoenix1991-lab** [project64/project64#1683]
Greetings, This is Analis. I am completely blind, and use a screen reader with Project 64. I have a problem with the cheats menu. When I go to enable a cheat, I cannot enable it. I am using Non Visual Desktop Access, or NVDA screen reader. I get three different options on eny given cheat, checked, half checked, or not checked. I have tried all three of these options to no avail. I have hide advanced settings disabled and remember selected cheats enabled. Do I need to do anything else? I have project 64 installed in a folder in which I have full permission to manipulate. Any help would be appreciated. Please get back to me ASAP! Thank you!

**Comment by IronYoshi64** [comment]
What is the version of Project64 are you using?

**Comment by Phoenix1991-lab** [comment]
I am using the latest version of project 64. I have followed the
instructions exactly as they appear on the forum and website and cannot get
those stupid cheats to work. Thank you!

**Comment by IronYoshi64** [comment]
@Phoenix1991-lab Latest version as in the stable build, which is 2.3.2.202, which is outdated now, or the latest development version as in 2.4.0-1077? Also in the Config folder, there is a file called "Project64.cht". This is the cheat database file. This file for you may be in Read-only mode? Try to right click it, go to Properties and see if the Read-only mode is enabled or not for you.

**Comment by Phoenix1991-lab** [comment]
I downloaded the latest version of project 64 and checked if that file is
read only. It is not read only. I tried the cheats again and still cannot
get them to work.

**Comment by IronYoshi64** [comment]
That is weird indeed. Hmm.....
Can you attach your "Project64.cfg" file from the Config folder here so I can see it please.
This file is where Project64 saves its configuration and settings for various things.

**Comment by Phoenix1991-lab** [comment]
Here is the config file along with the cht file. Hope that will help you.

**Comment by IronYoshi64** [comment]
Umm... I do not see those files here. Try to manually drag them, 1 at a time, from the Windows Explorer to the Write reply box here.

**Comment by Phoenix1991-lab** [comment]
Here you go. Hope you get them this time.

**Comment by IronYoshi64** [comment]
Still nothing... no files attached...
The replies that you are doing are just simple plain text messages.
If you can't do this, then maybe you can just open up the "Project64.cfg" file in Notepad, select all the text that is there, then paste all that text in a reply.

**Comment by Phoenix1991-lab** [comment]
Here is the config text below.
[Plugin]
Audio Dll Ver=Project64 Audio Plugin: 1.0.1.1077-gb78a8d1
Controller Dll Ver=N-Rage For PJ64: 2.5.3.1077-gb78a8d1
Graphics Dll Ver=Project64 Video Plugin: 2.2.0.1077-gb78a8d1
RSP Dll Ver=RSP Plugin 1.7.4.1077-gb78a8d1

[Settings]
Basic Mode=0
Current Language=English
Remember Cheats=1

[Support Project64]
Run Count=3

[Recent File]
Recent Rom 0=C:\Users\Analis\Documents\Super Smash Bros. (USA).z64

[Main Window]
Left=355
Top=114

[916B8B5B-780B85A4-C:45]
Good Name=Super Smash Bros. (U)
Hope that helps.

**Comment by IronYoshi64** [comment]
Yeah that worked, thanks. From what I see, nothing looks weird or out of the ordinary.
So from what I can see from the reply, the game you are trying to enable cheats for is Super Smash Bros, right?
I will be back on my main computer at home in about an hour and a half or so. When I get back, I will try to look into this further, and post here any solutions I can think of.

**Comment by IronYoshi64** [comment]
Anyways, can you tell me what game that the cheats would not work for you? And what cheats were they?

**Comment by Phoenix1991-lab** [comment]
I was playing Super Smash Bros. and the cheat I was trying to use was have
all characters.

**Comment by IronYoshi64** [comment]
That cheat should work that is already built in, I just did it, and yeah it works. Did you try to enable it, then immediately go into the VS mode and they should be there. If that is what you want. I can give you a save file that has all of the characters and the mushroom kingdom stage as an alternative if you can't get the cheats working.

**Comment by Phoenix1991-lab** [comment]
I am completely blind and cannot enable the cheat. When I try with NVDA,
which is the screen reader I am using, there are three options, checked,
half checked, and not checked. I have tried all three options to no avail.
Worse yet, when I try to enable the cheat, it resets itself when I close
the cheats dialogue. What do I have to do to enable the cheat?

**Comment by IronYoshi64** [comment]
To enable cheats, you need to have the check next to it, then just close the cheats dialogue box. It should immediately activate, depending on the cheat. Cheats for lots of games are in Tree grouping format, meaning like this: Have/Infinite/Health.
If you just want to have the unlockables unlocked for Smash Bros. I can just attach a Save file that you can download and put in your Save folder where the emulator is. That may be easier.

**Comment by Phoenix1991-lab** [comment]
I understand that, but what I am trying to tell you that according to my
screen reader, everything is checked when I open the cheats menu. When I
try to manipulate a cheat then close out of the menu, it resets itself.
What can I do to make the cheat menu appear correctly?

**Comment by IronYoshi64** [comment]
The cheat dialogue box is fine when it come up. You really just need to know exactly how it displays. Its hard to explain for me. I do not want to be rude, but you may not be able to do exactly what you want to do, or at least I do not know what else to tell you. Sorry. You may need to wait for someone that knows more about screen readers than I do. If you want, I can send you a Save file that has everything unlocked. Just let me know if you want the Save.

**Comment by DrSooom** [comment]
@Phoenix1991-lab: When the cheat dialog is open, press NVDA+B (desktop keyboard layout) and (hopefully) the whole screen should be spoken by NVDA. After you have moved the focus to an entry (= cheat), disable the NumPad and press NVDA+NumPad4 and NVDA+NumPad6 (also in the desktop keyboard layout). If the mouse tracking is enabled (NVDA+M), then please move the mouse within this window. Maybe you'll find some checkboxes in the same line of such a cheat entry. Please also take a look at the section ["5.7. Navigating with the Mouse" in [NVDA's User Guide](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#NavigatingWithTheMouse).

@IronYoshi64: Does Project64 use Qt? If yes, which version? There are a few accessibility bugs in recent Qt 5.12.x versions. Qt 5.9.8 (used by MuseScore 3.3.x) works fine with NVDA 2019.2.1.

NVDA uses MSAA/iAccessible2 and UIA to get the information of the objects on the screen. You can run [NVDA 2019.2.1](https://www.nvaccess.org/download/?nvdaVersion=2019.2.1) as a portable application, just open the Installer and create it. A temporary (portable) copy of NVDA is already created and running automatically when opening the Installer. This temporary copy of NVDA will be deleted after exiting NVDA via its systray menu (also called NVDA menu, NVDA+N) or existing the Installer window. The NVDA key I mentioned above is the INS and the NumPadINS key by default. But you can also use the CapsLock key as the NVDA modifier key. There are hundreds of hotkeys in NVDA, which are different between the desktop and the laptop keyboard layout. The desktop keyboard layout is the default one. If you can't understand the speech output, please open the Speech Viewer in the Tools submenu in the NVDA menu.

See also: nvaccess/nvda#10617 and https://musescore.org/en/node/285464
